solo-chains/runtime/starlight/src/lib.rs

```diff
@@ -1484,6 +1492,60 @@ parameter_types! {
     pub const MaxLengthParaIds: u32 = 100u32;
     pub const MaxEncodedGenesisDataSize: u32 = 5_000_000u32; // 5MB
 }
+
+pub struct InnerStarlightRegistrar<AccountId, RegistrarManager>(
+    PhantomData<(AccountId, RegistrarManager)>,
+);
+impl<AccountId, RegistrarManager> RegistrarHandler<AccountId>
+    for InnerStarlightRegistrar<AccountId, RegistrarManager>
+where
+    RegistrarManager: RegistrarInterface<AccountId = AccountId>,
+{
+    fn register(
+        who: AccountId,
+        id: ParaId,
+        genesis_storage: Vec<ContainerChainGenesisDataItem>,
+    ) -> DispatchResult {
+        // Build HeadData
+        let key_values: Vec<(Vec<u8>, Vec<u8>)> =
+            genesis_storage.into_iter().map(|x| x.into()).collect();
+        let genesis_head = HeadData(key_values.clone().encode());
+
+        // Check if the wasm code is present in storage
+        let kv_code = key_values
+            .into_iter()
+            .find(|kv| kv.0 == StorageWellKnownKeys::CODE.to_vec());
+
+        if let None = kv_code {
+            return Err(DispatchError::Other("Code not found"));
+        }
+
+        // Build ValidationCode
+        let validation_code = ValidationCode(kv_code.unwrap().1);
```

Comment:
You can rewrite this to avoid unwrap

```rust
if let Some((_, code)) = kv_code {
    // Build ValidationCode
    let validation_code = ValidationCode(code);
    RegistrarManager::register(who, id, genesis_head, validation_code)
} else {
    return Err(DispatchError::Other("Code not found"));
}
```

Reply:
fixed